### ISSUE TITLE
Why is the accuracy of my training always zero?

### ISSUE THREAD

**Issue by zhangluustb** [WenmuZhou/crnn.gluon#3]
Why is the accuracy of my training always zero?

**Comment by WenmuZhou** [comment]
Can you paste the log?

**Comment by zhangluustb** [comment]
2018-12-11 11:26:50 INFO      train.py: logger init finished
2018-12-11 11:26:50 INFO      train.py: train with gpu 1 and mxnet 1.3.0
4000
 0123456789
8
2018-12-11 11:26:51 INFO      train.py: each epoch contains 32 steps
[11:26:52] src/operator/nn/./cudnn/./cudnn_algoreg-inl.h:109: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
2018-12-11 11:27:01 INFO      train.py: [0/10], [9/32],step: 9, Speed: 131.471 samples/sec, ctc loss: 126.8844,acc: 0.0000, lr:0.001, time:9.7360 s
2018-12-11 11:27:05 INFO      train.py: [0/10], [19/32],step: 19, Speed: 269.333 samples/sec, ctc loss: 102.3870,acc: 0.0000, lr:0.001, time:4.7525 s
2018-12-11 11:27:10 INFO      train.py: [0/10], [29/32],step: 29, Speed: 269.179 samples/sec, ctc loss: 67.6291,acc: 0.0000, lr:0.001, time:4.7552 s
2018-12-11 11:27:12 INFO      train.py: start val ....
[11:27:13] src/operator/nn/./cudnn/./cudnn_algoreg-inl.h:109: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
2018-12-11 11:27:17 INFO      train.py: Epoch 0,train_acc 0.0000, val_acc 0.0000
2018-12-11 11:27:22 INFO      train.py: [1/10], [9/32],step: 41, Speed: 246.794 samples/sec, ctc loss: 41.2836,acc: 0.0000, lr:0.001, time:5.1865 s
2018-12-11 11:27:27 INFO      train.py: [1/10], [19/32],step: 51, Speed: 268.779 samples/sec, ctc loss: 34.6337,acc: 0.0000, lr:0.001, time:4.7623 s
2018-12-11 11:27:32 INFO      train.py: [1/10], [29/32],step: 61, Speed: 268.517 samples/sec, ctc loss: 33.7717,acc: 0.0000, lr:0.001, time:4.7669 s
2018-12-11 11:27:32 INFO      train.py: start val ....
2018-12-11 11:27:34 INFO      train.py: Epoch 1,train_acc 0.0000, val_acc 0.0000
2018-12-11 11:27:39 INFO      train.py: [2/10], [9/32],step: 73, Speed: 245.719 samples/sec, ctc loss: 33.2368,acc: 0.0000, lr:0.001, time:5.2092 s
2018-12-11 11:27:44 INFO      train.py: [2/10], [19/32],step: 83, Speed: 267.884 samples/sec, ctc loss: 33.3093,acc: 0.0000, lr:0.001, time:4.7782 s
2018-12-11 11:27:49 INFO      train.py: [2/10], [29/32],step: 93, Speed: 268.502 samples/sec, ctc loss: 33.2528,acc: 0.0000, lr:0.001, time:4.7672 s
2018-12-11 11:27:49 INFO      train.py: start val ....
2018-12-11 11:27:50 INFO      train.py: Epoch 2,train_acc 0.0000, val_acc 0.0000
2018-12-11 11:27:56 INFO      train.py: [3/10], [9/32],step: 105, Speed: 244.218 samples/sec, ctc loss: 33.3869,acc: 0.0000, lr:0.001, time:5.2412 s
2018-12-11 11:28:00 INFO      train.py: [3/10], [19/32],step: 115, Speed: 267.906 samples/sec, ctc loss: 33.0621,acc: 0.0000, lr:0.001, time:4.7778 s
2018-12-11 11:28:05 INFO      train.py: [3/10], [29/32],step: 125, Speed: 267.573 samples/sec, ctc loss: 33.1792,acc: 0.0000, lr:0.001, time:4.7837 s
2018-12-11 11:28:06 INFO      train.py: start val ....
2018-12-11 11:28:07 INFO      train.py: Epoch 3,train_acc 0.0000, val_acc 0.0000
2018-12-11 11:28:12 INFO      train.py: [4/10], [9/32],step: 137, Speed: 248.315 samples/sec, ctc loss: 33.3102,acc: 0.0000, lr:0.001, time:5.1548 s
2018-12-11 11:28:17 INFO      train.py: [4/10], [19/32],step: 147, Speed: 267.623 samples/sec, ctc loss: 33.1677,acc: 0.0000, lr:0.001, time:4.7828 s
2018-12-11 11:28:22 INFO      train.py: [4/10], [29/32],step: 157, Speed: 267.142 samples/sec, ctc loss: 33.0259,acc: 0.0000, lr:0.001, time:4.7915 s
2018-12-11 11:28:23 INFO      train.py: start val ....
2018-12-11 11:28:23 INFO      train.py: Epoch 4,train_acc 0.0000, val_acc 0.0000
2018-12-11 11:28:29 INFO      train.py: [5/10], [9/32],step: 169, Speed: 246.933 samples/sec, ctc loss: 33.1781,acc: 0.0000, lr:0.001, time:5.1836 s
2018-12-11 11:28:34 INFO      train.py: [5/10], [19/32],step: 179, Speed: 267.204 samples/sec, ctc loss: 32.9955,acc: 0.0000, lr:0.001, time:4.7904 s
2018-12-11 11:28:38 INFO      train.py: [5/10], [29/32],step: 189, Speed: 267.647 samples/sec, ctc loss: 33.2978,acc: 0.0000, lr:0.001, time:4.7824 s
2018-12-11 11:28:39 INFO      train.py: start val ....
2018-12-11 11:28:40 INFO      train.py: Epoch 5,train_acc 0.0000, val_acc 0.0000
2018-12-11 11:28:45 INFO      train.py: [6/10], [9/32],step: 201, Speed: 246.300 samples/sec, ctc loss: 33.1656,acc: 0.0000, lr:0.001, time:5.1969 s
2018-12-11 11:28:50 INFO      train.py: [6/10], [19/32],step: 211, Speed: 267.062 samples/sec, ctc loss: 33.1633,acc: 0.0000, lr:0.001, time:4.7929 s
2018-12-11 11:28:55 INFO      train.py: [6/10], [29/32],step: 221, Speed: 267.000 samples/sec, ctc loss: 33.0914,acc: 0.0000, lr:0.001, time:4.7940 s
2018-12-11 11:28:55 INFO      train.py: start val ....
2018-12-11 11:28:56 INFO      train.py: Epoch 6,train_acc 0.0000, val_acc 0.0000
2018-12-11 11:29:02 INFO      train.py: [7/10], [9/32],step: 233, Speed: 227.916 samples/sec, ctc loss: 33.2145,acc: 0.0000, lr:0.001, time:5.6161 s
2018-12-11 11:29:07 INFO      train.py: [7/10], [19/32],step: 243, Speed: 266.745 samples/sec, ctc loss: 32.9802,acc: 0.0000, lr:0.001, time:4.7986 s
2018-12-11 11:29:12 INFO      train.py: [7/10], [29/32],step: 253, Speed: 266.788 samples/sec, ctc loss: 33.2383,acc: 0.0000, lr:0.001, time:4.7978 s
2018-12-11 11:29:12 INFO      train.py: start val ....
2018-12-11 11:29:14 INFO      train.py: Epoch 7,train_acc 0.0000, val_acc 0.0000
2018-12-11 11:29:19 INFO      train.py: [8/10], [9/32],step: 265, Speed: 239.442 samples/sec, ctc loss: 33.1284,acc: 0.0000, lr:0.001, time:5.3458 s
2018-12-11 11:29:24 INFO      train.py: [8/10], [19/32],step: 275, Speed: 266.617 samples/sec, ctc loss: 33.1644,acc: 0.0000, lr:0.001, time:4.8009 s
2018-12-11 11:29:29 INFO      train.py: [8/10], [29/32],step: 285, Speed: 266.679 samples/sec, ctc loss: 33.0298,acc: 0.0000, lr:0.001, time:4.7998 s
2018-12-11 11:29:29 INFO      train.py: start val ....
2018-12-11 11:29:30 INFO      train.py: Epoch 8,train_acc 0.0000, val_acc 0.0000
2018-12-11 11:29:36 INFO      train.py: [9/10], [9/32],step: 297, Speed: 245.642 samples/sec, ctc loss: 33.1129,acc: 0.0000, lr:0.001, time:5.2108 s
2018-12-11 11:29:40 INFO      train.py: [9/10], [19/32],step: 307, Speed: 266.180 samples/sec, ctc loss: 32.9064,acc: 0.0000, lr:0.001, time:4.8088 s

**Comment by WenmuZhou** [comment]
as you see, the `ctc loss` is big when training stop, you should train more epochs.

**Comment by zhangluustb** [comment]
Thank you very much for your reply.
And,
1. Can the alphabet include the "blank space"?
2. What is the purpose of the 'resnet_pytorch/13_0.9237533417290824.pth' file? Can I ignore it?
3. How many epochs do I need without the pre-training?

**Comment by WenmuZhou** [comment]
[there](https://github.com/WenmuZhou/crnn.gluon/blob/master/keys.py#L15) are a example of alphabat, the `BLANK_SYMBOL` show be in the last index of your alphabet.

 the 'resnet_pytorch/13_0.9237533417290824.pth' in [config.py](https://github.com/WenmuZhou/crnn.gluon/blob/master/config.py#L30) is used when resume training, you can ignore it.

In my case, after 30 epochs, the accuracy began to increase
